packages/react-api/src/api/model.js

```diff
@@ -66,15 +66,46 @@ export function executeModel(props) {
     filtersLogicalOperator
   };
 
-  // API supports multiple filters, we apply it only to geoColumn
-  const spatialFilters = spatialFilter
-    ? {
-        [source.geoColumn ? source.geoColumn : DEFAULT_GEO_COLUMN]: spatialFilter
+  let spatialFilters;
+  if (spatialFilter) {
+    let spatialDataType = source.spatialDataType;
+    let spatialDataColumn = source.spatialDataColumn;
+
+    if (!spatialDataType || !spatialDataColumn) {
+      if (source.geoColumn) {
+        const parsedGeoColumn = source.geoColumn ? source.geoColumn.split(':') : [];
+        if (parsedGeoColumn.length === 2) {
+          spatialDataType = parsedGeoColumn[0];
+          spatialDataColumn = parsedGeoColumn[1];
+        } else if (parsedGeoColumn.length === 1) {
+          spatialDataColumn = parsedGeoColumn[0] || DEFAULT_GEO_COLUMN;
+          spatialDataType = 'geo';
+        }
+        if (spatialDataType === 'geom') {
+          // fallback if for some reason someone provided old `geom:whatever`
+          spatialDataType = 'geo';
+        }
+      } else {
+        spatialDataType = 'geo';
+        spatialDataColumn = DEFAULT_GEO_COLUMN;
       }
-    : undefined;
+    }
+
+    // API supports multiple filters, we apply it only to geometry column or spatialDataColumn
+    spatialFilters = spatialFilter
+      ? {
+          [spatialDataColumn]: spatialFilter
+        }
+      : undefined;
 
-  if (spatialFilters) {
     queryParams.spatialFilters = JSON.stringify(spatialFilters);
+    queryParams.spatialDataType = spatialDataType;
+    if (spatialDataType !== 'geo') {
+      if (source.spatialFiltersResolution !== undefined) {
```

Comment:
Out of curiosity: Could `source.spatialFiltersResolution` be `null` at some point? If so, maybe we can double negate this expression to prevent this case, or is there any edge case related to a 0 resolution?

Reply:
Well, this we can't protect user from themselves.

If he passed number, he's ok. If he didn't passed anything (undefined), he's ok - default is used. If he passed null or object or date, it's his own fault - i don't believe that we should check _every_ possible failure mode if we already defined that `spatialFiltersResolution` is `optional number`.

This is just API wrapper, nothing will explode - just weird stuff will be sent to.

(and btw, this is internal code - validation if required should be placed somewhere on top of widget)

Reply:
I was thinking more about something like `!!source.spatialFiltersResolution`, WDYT?

Reply:
I don't like `!!source.spatialFiltersResolution` for one reason: who said, 0 is bad value for this parameter (is it?). I only care if user provided it or not - and that's why we have `undefined` in JS.